presto-main/src/main/java/com/facebook/presto/server/thrift/ThriftServerInfoClient.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.server.thrift;
+
+import com.facebook.drift.annotations.ThriftMethod;
+import com.facebook.drift.annotations.ThriftService;
+import com.google.common.util.concurrent.ListenableFuture;
+
+@ThriftService("ThriftServerInfoService")
+public interface ThriftServerInfoClient
+{
+    /**
+     * Use integers to represent the ordinal of the enum.
+     * Use NodeState::valueOf to recover the enum from an integer
```

Comment:
Ideally you could use `ThriftEnum`?

Reply:
Ya, but unfortunately the class is in SPI... so I took the workaround